### PR TITLE
Highlight digits in atoms

### LINKED ISSUE #172: Numeric digits in atoms break syntax highlighting
For example, the atom `:msisdn2pid` which comes from an example in a book I'm reading.

screenshot 2015-03-02 11 38 01

### PR DESCRIPTION
**Pull request by rranelli**
Fixes #172

### CODE REVIEW

File: elixir-mode.el [hunk 1]
```diff
@@ -185,7 +185,9 @@ for the Elixir programming language."
     `(
       (atoms . ,(rx ":"
                     (or
-                     (one-or-more (any "a-z" "A-Z" "_" "\"" "'"))
+                     (and
+                      (any "a-z" "A-Z" "_" "\"" "'")
+                      (zero-or-more (any "a-z" "A-Z" "0-9" "_" "\"" "'")))
```
Comment (mattdeboard) [inline elixir-mode.el]: Why the change from `one-or-more` to `zero-or-more`? This doesn't make sense to me.

Comment (rranelli) [inline elixir-mode.el]: The old regexp was:

```lisp
(or 
  (one-or-more (any "a-z" "A-Z" "_" "\"" "'"))

...)
```

Which matches only `:msisdn` in the `:msisdn2pid` atom.

By changing it to 

```lisp
(or 
  (and
    (any "a-z" "A-Z" "_" "\"" "'")
    (zero-or-more (any "a-z" "A-Z" "0-9" "_" "\"" "'")))
```

I ask for the first character in `'("a-z" "A-Z" "_" "\"" "'")` and the subsequent ones in `'("a-z" "A-Z" "0-9" "_" "\"" "'")`. This in turn matches the whole `:msisdn2pid` atom.

Sorry, I should've added more context to the PR.

Comment (mattdeboard) [inline elixir-mode.el]: Makes sense, thanks

**Comment by mattdeboard** [comment]
Please just add a regression test for this change and it should be good to merge.

**Comment by rranelli** [comment]
Alright, will do.

**Comment by mattdeboard** [comment]
You can see an example [here](https://github.com/elixir-lang/emacs-elixir/blob/master/test/elixir-mode-font-test.el#L31-L44).

**Comment by rranelli** [comment]
@mattdeboard I have added the regression test for #172 as you requested.

(also, why do we run `cask test --no-win` in travis? I don't understand the need or the advantages of it)

**Comment by andreas-roehler** [comment]
@mattdeboard  ert itself has some bugs WRT to display and buffer. See here for example:

http://debbugs.gnu.org/db/16/16853.html

Suggest to drop these --no-win tests.

**Comment by rranelli** [comment]
ping,

Any more thoughts on this PR?

Thanks!
